### PR TITLE
Some minor fix

### PR DESCRIPTION
**Pull request by critical27**
## What type of PR is this?
- [ ] bug
- [ ] feature
- [X] enhancement

#### Description:
Minor fix:
1. ScanNode has not handle filter correctly in community version, ent version has been modified before
2. Add reserve before return when storage run concurrently

## Checklist:
Tests:
- [X] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Review by Shylock-Hg — APPROVED** [review]

**Review by SuperYoko — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]

File: src/storage/query/ScanEdgeProcessor.cpp [hunk 1]
```diff
@@ -179,6 +179,11 @@ void ScanEdgeProcessor::runInMultipleThread(const cpp2::ScanEdgeRequest& req) {
   folly::collectAll(futures).via(executor_).thenTry([this](auto&& t) mutable {
     CHECK(!t.hasException());
     const auto& tries = t.value();
+    size_t sum = 0;
+    for (size_t j = 0; j < tries.size(); j++) {
+      sum += results_[j].size();
+    }
+    resultDataSet_.rows.reserve(sum);
     for (size_t j = 0; j < tries.size(); j++) {
       CHECK(!tries[j].hasException());
```
Comment (SuperYoko) [inline src/storage/query/ScanEdgeProcessor.cpp:188]: Why do your remove `CHECK(!tries[j].hasException());` in L133 in GetPropProcessor.cpp while not here?

File: src/storage/exec/ScanNode.h [hunk 2]
```diff
@@ -171,9 +171,15 @@ class ScanVertexPropNode : public QueryNode<Cursor> {
...
         }
       }
-      if (ret == nebula::cpp2::ErrorCode::SUCCEEDED &&
-          (filter_ == nullptr || QueryUtils::vTrue(filter_->eval(*expCtx_)))) {
-        resultDataSet_->rows.emplace_back(std::move(row));
+      if (ret == nebula::cpp2::ErrorCode::SUCCEEDED) {
+        if (filter_ == nullptr) {
+          resultDataSet_->rows.emplace_back(std::move(row));
+        } else {
+          auto result = QueryUtils::vTrue(filter_->eval(*expCtx_));
```
Comment (panda-sheep) [inline src/storage/exec/ScanNode.h:178]: The return type of` QueryUtils::vTrue` has changed

Comment (critical27) [reply src/storage/exec/ScanNode.h:178]: Good catch, my bad …… I'll keep code in community version same as ent version

File: src/storage/query/ScanEdgeProcessor.cpp (hunk 1 above)
Comment (critical27) [reply src/storage/query/ScanEdgeProcessor.cpp:188]: 👌 , will keep consistent~

**Pull request closed without merging**